packages/next/next-server/server/load-components.ts

```diff
@@ -16,12 +18,20 @@ export type LoadComponentsReturnType = {
   Component: any
   pageConfig: PageConfig
   unstable_getStaticProps?: (params: {
-    params: any
+    params: { [key: string]: string | string[] }
   }) => {
     props: any
     revalidate?: number | boolean
   }
-  unstable_getStaticPaths?: () => void
+  unstable_getStaticPaths?: () => Promise<
+    Array<string | { [key: string]: string | string[] }>
```

Comment:
Want to add all these types in a diff PR?

Reply:
Opened PR here: https://github.com/zeit/next.js/pull/10259